.github/workflows/tsbench-coverage.yaml

```diff
@@ -0,0 +1,204 @@
+# A job to inspect code coverage of current TSBench benchmarks, and also run
+# them with assertions enabled to have more test coverage.
+name: TSBench Code Coverage
+"on":
+  schedule:
+    - cron: '0 3 * * *'
+  push:
+    branches:
+      - main
+      - trigger/tsb-coverage
+  pull_request:
+    paths: .github/workflows/tsbench-coverage.yaml
+
```

Comment:
Is there a specific reason why we don't allow `workflow_dispatch` here? Might be useful to be able to trigger the workflow directly from the GitHub UI.